src/EditorFeatures/CSharpTest/RemoveUnusedMembers/RemoveUnusedMembersTests.cs

```diff
@@ -0,0 +1,1475 @@
+        public async Task IndexerIsRead()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|this|][int x] { get { return 0; } set { } }
+    public int M(int x) => this[x];
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EventIsRead()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System;
+
+class MyClass
+{
+    private event EventHandler [|e|];
+    public EventHandler P => e;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EventIsSubscribed()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System;
+
+class MyClass
+{
+    private event EventHandler [|e|];
+    public void M()
+    {
+        e += MyHandler;
+    }
+
+    static void MyHandler(object sender, EventArgs e)
+    {
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EventIsRaised()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System;
+
+class MyClass
+{
+    private event EventHandler [|_eventHandler|];
+
+    public void RaiseEvent(EventArgs e)
+    {
+        _eventHandler(this, e);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldInNameOf()
+        {
+            // https://github.com/dotnet/roslyn/issues/29519 tracks flagging and fixing this appropriately.
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    private string _goo2 = nameof(_goo);
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldInDocComment()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+/// <summary>
+/// <see cref=""C._goo""/>
+/// </summary>
+class C
+{
+    private static int [|_goo|];
+}",
+@"
+/// <summary>
+/// <see cref=""C._goo""/>
+/// </summary>
+class C
+{
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsOnlyWritten()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    public void M()
+    {
+        _goo = 0;
+    }
+}",
+@"class MyClass
+{
+    public void M()
+    {
+        _goo = 0;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task PropertyIsOnlyWritten()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|P|] { get; set; }
+    public void M()
+    {
+        P = 0;
+    }
+}",
+@"class MyClass
+{
+    public void M()
+    {
+        P = 0;
```

Comment:
Code was also broken.  Users will assume if we're telling them that the value is not used that they can safely remove, and that we will do so in a way that will leave the code compiling and working properly.